test/src/sdk-harness/test_projects/token_ops/mod.rs

```diff
@@ -387,11 +387,8 @@ async fn can_send_message_output_with_data() {
     assert_eq!(*fuelcoin_id, **message_receipt.sender().unwrap());
     assert_eq!(&recipient_address, message_receipt.recipient().unwrap());
     assert_eq!(amount, message_receipt.amount().unwrap());
-    assert_eq!(24, message_receipt.len().unwrap());
-    assert_eq!(
-        vec![0, 0, 0, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 0, 75, 0, 0, 0, 0, 0, 0, 0, 50],
-        message_receipt.data().unwrap()
-    );
+    assert_eq!(8, message_receipt.len().unwrap());
+    assert_eq!(vec![100, 75, 50], message_receipt.data().unwrap());
```

Comment:
If the type is changed why is the test also changed? 
Shouldn't this be the same as before?

I see that the `token_ops` has changed so I assume that has something to do with it.

Reply:
This is because of the difference in the way `u8` and `u64` are encoded in Rust and Sway.
In Sway, both are 8 bytes long, whereas in Rust a `u8` is only a single byte.
Previously the tests were passing in `vec![100, 75, 50]` as the data. This was a `Vec<u64>` in sway, and so we had to expect the padded `vec![0, 0, 0, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 0, 75, 0, 0, 0, 0, 0, 0, 0, 50]` in our assertion.
Now that the data has the type `Bytes`, there is no more padding and so we can assert using the unpadded 
`vec![100, 75, 50]`.